packages/webapp/src/containers/SensorReadings/saga.js

```diff
@@ -205,30 +205,39 @@ export function* getSensorsReadingsSaga({ payload }) {
         }
         sensorReadingData[TEMPERATURE] = Object.values(ambientDataWithSensorsReadings);
       } else if (readingType === SOIL_WATER_POTENTIAL) {
-        const soilWaterPotentialReadings = result?.data?.sensorReading[SOIL_WATER_POTENTIAL].reduce(
-          (acc, cv) => {
-            const dt = new Date(cv?.read_time).getTime() / 1000;
-            let dateAndTimeInfo = new Date(dt * 1000).toString();
-            const isCorrectTimestamp = GRAPH_TIMESTAMPS?.find((g) => dateAndTimeInfo?.includes(g));
-            if (isCorrectTimestamp && startDate < dt && dt < predictedEndDate) {
-              const currentDateTime = `${dateAndTimeInfo?.split(':00:00')[0]}:00`;
-              if (!acc[dt]) acc[dt] = {};
-              acc[dt] = {
-                ...acc[dt],
-                [cv?.name]: getSoilWaterPotentialValue(cv?.value, measurement),
-                [CURRENT_DATE_TIME]: currentDateTime,
-              };
-            }
-            return acc;
-          },
-          {},
-        );
+        let soilWaterPotentialReadings = [];
+        if (!result?.data?.sensorReading[SOIL_WATER_POTENTIAL].every((cv) => cv.value === null)) {
+          soilWaterPotentialReadings = result?.data?.sensorReading[SOIL_WATER_POTENTIAL].reduce(
+            (acc, cv) => {
+              const dt = new Date(cv?.read_time).getTime() / 1000;
+              let dateAndTimeInfo = new Date(dt * 1000).toString();
+              const isCorrectTimestamp = GRAPH_TIMESTAMPS?.find((g) =>
+                dateAndTimeInfo?.includes(g),
+              );
+              if (isCorrectTimestamp && startDate < dt && dt < predictedEndDate) {
+                const currentDateTime = `${dateAndTimeInfo?.split(':00:00')[0]}:00`;
 
-        lastUpdatedReadingsTime[SOIL_WATER_POTENTIAL] = getLastUpdatedTime(
-          Object.keys(soilWaterPotentialReadings),
-        );
+                if (!isFound && currentDT < dt) {
+                  isFound = true;
+                  predictedXAxisLabel = acc[Object.keys(acc).at(-1)][CURRENT_DATE_TIME];
+                }
 
-        const soilWaterPotentialSensorReadings = Object.values(soilWaterPotentialReadings);
+                if (!acc[dt]) acc[dt] = {};
+                acc[dt] = {
+                  ...acc[dt],
+                  [cv?.name]: isNaN(getSoilWaterPotentialValue(cv?.value, measurement))
+                    ? i18n.t('translation:SENSOR.NO_DATA')
+                    : getSoilWaterPotentialValue(cv?.value, measurement),
+                  [CURRENT_DATE_TIME]: currentDateTime,
+                };
```

Comment:
Would this still work even if `[cv?.name]` is undefined?

Reply:
Its really just a re-organization of what was originally here. I did not re-write any of this. I think this component needs an overall refactor.

If I am looking critically at this:
To me cv? is useless because it should always be true -- why would reduce() iterate over something that doesn't exist. 

And cv.name should also always be defined since locationsModel enforces new locations(including sensors) to have a name.